File: 103-2_iOS程式設計/DatePickerDemo/DatePickerDemo/ViewController.m
//
//  ViewController.m
//  DatePickerDemo
//
//  Created by CSIEVNUT403 on 2015/3/23.
//  Copyright (c) 2015年 CSIEVNUT403. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UILabel *label;

- (IBAction)selectDate:(id)sender;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)selectDate:(id)sender {
    UIDatePicker *datePicker = (UIDatePicker *) sender;
    
    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
    [dateFormatter setDateFormat:@"yyyy/MM/dd hh:mm:ss"];
    
    NSString *dateString = [dateFormatter stringFromDate:[datePicker date]];
    
    [self.label setText:dateString];
}

@end

File: 106-2_iOS程式設計/DatePicker/DatePicker/ViewController.swift
//
//  ViewController.swift
//  DatePicker
//
//  Created by Brian Wang on 2018/5/31.
//  Copyright © 2018年 Brian Wang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var datePicker: UIDatePicker!
    
    @IBOutlet weak var labelMsg: UILabel!
    
    var dateFormatter = DateFormatter()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        datePicker.datePickerMode = UIDatePickerMode.dateAndTime
        datePicker.locale = Locale(identifier: "zh_TW")
        datePicker.date = Date()
        dateFormatter.dateFormat = "西元y年M月d日 hh：mm：ss"
        labelMsg.text = dateFormatter.string(from: datePicker.date)
    }
    
    @IBAction func dateChange(_ sender: UIDatePicker) {
        labelMsg.text = dateFormatter.string(from: datePicker.date)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

